scipy/stats/tests/test_multivariate.py

```diff
@@ -1226,6 +1237,59 @@ def test_fit_fix_cov(self):
                                                      ).sum()
         assert logp_perturbed < logp_fix
 
+    @pytest.mark.parametrize('X_ndim', [3])
+    @pytest.mark.parametrize('dimensions', [[1], 
+                                            [0, 1]])
+    @pytest.mark.parametrize('frozen', [True, False])
+    @pytest.mark.parametrize('cov_object', [True, False])
+    def test_marginal_distribution(self, X_ndim, dimensions, frozen, cov_object):
+        rng = np.random.default_rng(413911473)
+        mean = rng.standard_normal(X_ndim)
+        A = rng.standard_normal((X_ndim, X_ndim))
+        cov = np.dot(A, A.transpose())
+        
+        if cov_object:
+            cov = _covariance.CovViaPrecision(cov)
+
+        x = np.random.standard_normal((4, len(dimensions))) # number of points at which to evaluate marginal PDF
+        X = multivariate_normal(mean, cov)  # frozen distribution
+        
+        if frozen:
+            Y = X.marginal(dimensions)
+            res = Y.pdf(x)
+        else: 
+            Y = multivariate_normal.marginal(dimensions, mean, cov)
+            res = Y.pdf(x)
+
+        ref = marginal_pdf(X, X_ndim, dimensions, x)
+        assert_allclose(ref, res)
+
+    def test_marginal_dimensions(self):
```

Comment:
A common name for this sort of test might be `test_input_validation`. Another test, `test_special_cases`, might be good for testing e.g. all dimensions retained / none marginalized out.